draft-barnes-acme.md

```diff
@@ -1093,6 +1093,26 @@ type (required, string):
 To validate a DNS challenge, the server queries for TXT records under the validation domain name.  If it receives a record whose contents match the token in the challenge, then the validation succeeds.  Otherwise, the validation fails.
 
 
+## Wildcard Domain Validation
+
+When the identifier being validated is a domain name, it may contain a wildcard character "\*" as its left-most label (e.g., "\*.foo.example.com"). A wildcard certificate contains a wildcard domain in a CN or subjectAltName of type DNS-ID and automatically vouches for any and all host names within the bare domain name (the FQDN to the immediate right of the wildcard label).
+
+Domain validation for issuance of a wildcard certificate proceeds as follows:
+1. The server MUST validate that the identifier contains a single wildcard character and that the wildcard character is the left-most label. For instance, "foo.\*.example.com" is invalid.
+2. The server MUST validate that the wildcard label does not fall immediately to the left of a registry-controlled or public suffix. For instance, "\*.co.uk" is invalid.
```

Comment:
The PSL currently contains the IANA Root Zone Database *PLUS* the domain boundaries defined by registries (e.g. .co.uk). "registry controlled" covers the former, "public suffix" covers the latter.

Historically, every gTLD was implicitly a PSL, but the PSL didn't express that (it instead let it fall back to the default '*' role described in the PSL algorithm). However, as a result of implementations coming to assume that the PSL >= the RZD, it's now been automated that the RZD data feed is incorporated and expressed in the PSL.

So yes, using a single datafile will work. However, if you want more timely security, the RZD is published multiple times a day (... insecurely, naturally), or you could execute a DNS zone file transfer multiple times a day. This is because PSL updates tend to be batched every month or two.

In either event, if a gTLD is absent from the PSL, MUST NOT issue is a correct response, period. Treat it as an Intranet domain until the PSL refreshes from the RZD.